moveit_core/trajectory_processing/src/time_optimal_trajectory_generation.cpp

```diff
@@ -110,7 +110,6 @@ class CircularPathSegment : public PathSegment
     const Eigen::VectorXd start_direction = (intersection - start).normalized();
```

Comment:
If these vectors are close to zero, normalization will result in NaNs or Infs. Not sure though, that this can happen.

Reply:
@rhaschke True, I suppose there is division in there.  I believe the following loop, which is done before the constructor for `Path`, ensures that the vectors are large enough that we do not have to worry about failure when taking the dot product.  On the other hand, I think once it gets into the heart of the algorithm there are various parts that are not robust to very small values, where for example a 180 degree turn, which this prevents from throwing an exception, might still result in very small time steps between switching points, causing failure during one of the integration methods. 
```
for (size_t j = 0; j < num_joints; j++)
{
  new_point[j] = waypoint->getVariablePosition(idx[j]);
  if (p > 0 && std::abs(new_point[j] - points.back()[j]) > 0.001)
    diverse_point = true;
}
```